src/components/DroppableContainer.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { RefObject, useRef } from "react";
+import { ConnectDropTarget, DropTargetMonitor, useDrop } from "react-dnd";
+import { dnd } from "../dnd";
+import { deselectAllChildren, useTemporaryView } from "./TemporaryViewContext";
+import { usePlanner } from "../planner";
+import {
+  ViewPath,
+  getParentKey,
+  useViewPath,
+  viewPathToString,
+} from "../ViewContext";
+import { NOTE_TYPE } from "./Node";
+
+export type DragItemType = {
+  path: ViewPath;
+};
+
+type DroppableContainerProps = {
+  children: React.ReactNode;
+};
+
+function calcDragDirection(
+  ref: RefObject<HTMLElement>,
+  monitor: DropTargetMonitor<DragItemType>,
+  path: ViewPath
+): number | undefined {
+  if (!monitor.isOver({ shallow: true })) {
+    return undefined;
+  }
+  if (!ref.current) {
+    return undefined;
+  }
+  const item = monitor.getItem();
+  if (item && viewPathToString(item.path) === viewPathToString(path)) {
+    return undefined;
+  }
+  const hoverBoundingRect = ref.current.getBoundingClientRect();
+  const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+  const clientOffset = monitor.getClientOffset();
+  if (!clientOffset || clientOffset.y === undefined) {
+    // This should only happen in test environment, therefore we assume dragging upwards
+    return 1;
+  }
+  const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+  // Dragging upwards
+  if (hoverClientY < hoverMiddleY) {
+    return 1;
+  }
+  // Dragging downwards
+  if (hoverClientY > hoverMiddleY) {
+    return -1;
+  }
+  return undefined;
+}
+
+function calcIndex(
+  index: number | undefined,
+  direction: number | undefined
+): number | undefined {
+  if (index === undefined || direction === undefined) {
+    return undefined;
+  }
+  return direction === 1 ? index : index + 1;
```

Comment:
sure it shouldn't be `index -1` when direction is upwards?

Reply:
When a Note is dragged over item 0, the Notes index should become 0 and what was 0 becomes 1. A -1 index can never be correct.

Reply:
got it. thanks. it works, so it must be correct ;)